packages/react-widgets/src/widgets/TimeSeriesWidget.js

```diff
@@ -0,0 +1,303 @@
+import React, { useEffect, useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { getTimeSeries } from '../models';
+import { addFilter, removeFilter, selectSourceById } from '@carto/react-redux';
+import { NoDataAlert, TimeSeriesWidgetUI, WrapperWidgetUI } from '@carto/react-ui';
+import {
+  GroupDateTypes,
+  AggregationTypes,
+  _FilterTypes as FilterTypes,
+  _getApplicableFilters as getApplicableFilters
+} from '@carto/react-core';
+import { capitalize, Menu, MenuItem, SvgIcon, Typography } from '@material-ui/core';
+import { TIME_SERIES_CHART_TYPES } from '@carto/react-ui';
+import { PropTypes } from 'prop-types';
+
+const BUCKET_SIZE_RANGE_MAPPING = {
+  [GroupDateTypes.WEEKS]: 60 * 60 * 24 * 7,
+  [GroupDateTypes.DAYS]: 60 * 60 * 24
+};
+
+function TimeSeriesWidget({
+  // Widget
+  id,
+  title,
+  dataSource,
+  column,
+  operationColumn,
+  operation,
+  stepSizeOptions,
+  onError,
+  wrapperProps,
+  noDataAlertProps,
+  // UI
+  chartType,
+  duration,
+  tooltip,
+  tooltipFormatter,
+  formatter,
+  height,
+  showControls,
+  isPlaying,
+  onPlay,
+  isPaused,
+  onPause,
+  onStop,
+  timelinePosition,
+  onTimelineUpdate,
+  timeframe,
+  onTimeframeUpdate,
+  // Both
+  stepSize
+}) {
+  const dispatch = useDispatch();
+  const [timeSeriesData, setTimeSeriesData] = useState([]);
+  const viewportFeaturesReady = useSelector((state) => state.carto.viewportFeaturesReady);
+  const [selectedStepSize, setSelectedStepSize] = useState(stepSize);
+
+  useEffect(() => {
+    if (stepSize !== selectedStepSize) {
+      setSelectedStepSize(stepSize);
+    }
+    // Only on bucketSize update
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [stepSize]);
+
+  const [isLoading, setIsLoading] = useState(false);
+
+  const filters = useSelector((state) => {
+    const source = selectSourceById(state, dataSource) || {};
+    const filters = { ...source.filters };
```

Comment:
Other widgets should do it also. Each time you select a category in CategoryWidget, the useEffect to get the data to feed the widget is being executed again, to obtain the exact same result because the filter that made the useEffect to update isn't being applied to the data. In the CategoryWidget it doesn't matter because you don't have 60 filters per second, but in TimeSeries yes.

Reply:
Not applicable anymore